File: src/main/java/ru/sicampus/bootcamp2026/dto/request/CreateMeetingRequest.java
package ru.sicampus.bootcamp2026.dto.request;

import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateMeetingRequest {

    @NotBlank(message = "Название встречи обязательно")
    @Size(max = 50, message = "Название встречи не должно превышать 50 символов")
    private String title;

    @Size(max = 2048, message = "Описание не должно превышать 2048 символов")
    private String description;

    @Size(max = 255, message = "Местоположение не должно превышать 255 символов")
    private String location;

    @NotNull(message = "Время начала обязательно")
    private Instant startTime;

    @NotNull(message = "Время окончания обязательно")
    private Instant endTime;

    @Size(min = 1, message = "Должен быть хотя бы один участник")
    @NotNull(message = "Список участников обязателен")
    private List<UUID> participantIds;
}

File: server/src/main/kotlin/com/nikshet/industrialbackend/dto/request/CreateMeetingRequest.kt
package com.nikshet.industrialbackend.dto.request

import jakarta.validation.constraints.*
import java.time.Instant
import java.time.ZoneOffset
import java.util.*

data class CreateMeetingRequest(
    @field:NotBlank
    @field:Size(max = 200)
    val title: String,

    @field:Size(max = 1000)
    val description: String? = null,

    @field:NotBlank
    @field:Size(max = 200)
    val location: String,

    @field:NotNull
    var startTime: Instant,

    @field:Min(1)
    @field:Max(8)
    val durationHours: Short,

    @field:NotEmpty
    val invitees: List<UUID>
) {
    init {
        val utcTime = startTime.atOffset(ZoneOffset.UTC).toLocalDateTime()
        require(utcTime.minute == 0 && utcTime.second == 0 && utcTime.nano == 0) {
            "Start time must be on the hour (e.g., 09:00, not 09:15). Got: $startTime"
        }
    }
}